### PR TITLE
Sqlcommenter integration into SQLAlchemy

### PR DESCRIPTION
**Pull request by Thiyagu55**
# Description

This PR integrates SQLCommenter into SQLAlchemy. When SQLCommenter is enabled in the library the query logs will have traceparent id appended

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
The test cases are tested using the below commands
`tox -e  py39-test-instrumentation-sqlalchemy14`  & `tox -e  py39-test-instrumentation-sqlalchemy14`
`

Unit Testcases has been added
- [X] test_sqlcommenter_enabled
- [X] test_sqlcommenter_disabled
- [X] test_generate_commenter

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [ ] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [X] Changelogs have been updated
- [X] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py [hunk 1]
```diff
@@ -115,6 +117,13 @@ def _before_cur_exec(
                     span.set_attribute(key, value)
 
         context._otel_span = span
+        if self.enable_commenter:
```
Comment (sjs994) [inline instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py:120]: Can we add some tests to validate this change ?

**Review by sjs994 — COMMENTED** [review]
Thanks  @Thiyagu55

File: instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py [hunk 2]
```diff
@@ -70,12 +71,13 @@ def _wrap_create_engine_internal(func, module, args, kwargs):
 
 
 class EngineTracer:
-    def __init__(self, tracer, engine):
+    def __init__(self, tracer, engine, enable_commenter):
```
Comment (lonewolf3739) [inline instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py:74]:
```suggestion
    def __init__(self, tracer, engine, enable_commenter = False):
```

File: instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py [hunk 3]
```diff
@@ -115,6 +117,13 @@ def _before_cur_exec(
                     span.set_attribute(key, value)
 
         context._otel_span = span
+        if self.enable_commenter:
+            _version = '00'
+            _span_id = opentelemetry.trace.format_span_id(span.context.span_id)
+            _trace_id = opentelemetry.trace.format_trace_id(span.context.trace_id)
+            _flags = str(opentelemetry.trace.TraceFlags.SAMPLED)
+            statement = statement + "/*traceparent=" + _version + '-' + _trace_id + '-' + _span_id + '-' + _flags + "*/"
```
[the remark this answers is not in the document]
Comment (lonewolf3739) [reply instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py:125]: @Thiyagu55 you can update the `_generate_sql_comment` to pass the whole span context and format in compliance with commenter spec.

File: instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/version.py [hunk 4]
```diff
@@ -12,4 +12,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = "0.28b1"
+__version__ = "0.28b2"
```
Comment (lonewolf3739) [inline instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/version.py:15]: You don't need to modify this, maintainers will bump it when they release a new version

Comment (Thiyagu55) [reply instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/version.py]: Reverted the changes

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased](https://github.com/open-telemetry/opentelemetry-python/compare/v1.9.1-0.28b1...HEAD)
 
+### Added
+
+ - `opentelemetry-instrumentation-sqlalchemy` added experimental sql commenter capability
+   ([#924](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/924/)
+
 ### Added
```
Comment (srikanthccv) [inline CHANGELOG.md:16]:
```suggestion
### Added

 - `opentelemetry-instrumentation-sqlalchemy` added experimental sql commenter capability
   ([#924](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/924/)
```

**Review by srikanthccv — APPROVED** [review]

**Review by sjs994 — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased](https://github.com/open-telemetry/opentelemetry-python/compare/v1.9.1-0.28b1...HEAD)
 
+### Added
+
+ - `opentelemetry-instrumentation-sqlalchemy` added experimental sql commenter capability
+   ([#924](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/924))
+
 ### Added
 
 - `opentelemetry-instrumentation-dbapi` add experimental sql commenter capability
```
Comment (srikanthccv) [inline CHANGELOG.md:17]: There are two `Added` sections here
```suggestion

### Added

- `opentelemetry-instrumentation-sqlalchemy` added experimental sql commenter capability
   ([#924](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/924))
- `opentelemetry-instrumentation-dbapi` add experimental sql commenter capability
```

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**